### DISCUSSION TITLE
多集群部署如何仅升级某一个集群的image 版本？如何删除一个集群上的部署？

### DISCUSSION THREAD

**Discussion by skyshaokai** [kubevela/kubevela#3705] [Q&A] [answered]
在我们的使用场景下，应用有多个环境部署。计划采用多集群部署https://kubevela.net/zh/docs/case-studies/multi-cluster
使用多 policies.topology 完成app 下部署多套环境。
1、多套线下环境，多个rd并行开发测试。
2、多套线上环境，高可用部署。

为了避免更新某个测试环境影响其他测试环境或者线上环境，希望可以仅更新某个测试环境。如何仅更新某套测试环境？
由于app 的部署情况可能会发生调整，例如某个测试环境测试完成后释放某套测试环境。如何才能删除某个环境？

**Comment by Somefive** [comment, marked as answer]
1. 测试环境和线上环境分别配置两个 Topology Policy T1 和 T2。
2. 测试环境和线上环境分别配置两个 Override Policy O1 和 O2，在 Override Policy 里面用 container-image trait 配置镜像。
3. 在工作流里面写两个 deploy 步骤，分别 T1+O1 以及 T2+O2。
4. 更新测试环境的时候更新 O1，然后触发应用重新部署。
5. 执行 T1+O1 的 deploy 步骤时会更新测试环境。
6. 执行 T2+O2 的 deploy 步骤时会发现线上环境没有变化所以不会做变动。

但是这个流程的潜在风险是，如果配置应用时不小心配错成了 T2/O2，那么这个变更就会发布到线上环境中。如果想避免这种误操作，可以将测试环境和线上环境拆分成两个应用；如果不存在这种误操作，可以不拆分。
